### PR TITLE
feat: flat search input value

### LINKED ISSUE #60: Add support for the new flat GraphQL search input value field
In the current version of the `graphql-server-model-codegen`,  any `type Query` in the generated schema that uses `readAll` and `readAllCursor` model methods, support a `search` input that takes the following shape:

```gql
input search<Model>Input {
    field: <Model>Field
    value: typeValue
    operator: Operator
    search: [search<Model>Input]
}
```
The `value` field is a composite object that required some redundancy when writing the query:

```gql
input typeValue: {
  type: String
  value: String!
}
```

To eliminate this redundancy and improve the query shape, in the next `graphql-server-model-codegen` version ([issue](https://github.com/Zendro-dev/graphql-server-model-codegen/issues/140) | [branch](https://github.com/Zendro-dev/graphql-server-model-codegen/commits/issue%23140-refactor-search-input) | [PR](https://github.com/Zendro-dev/graphql-server-model-codegen/pull/144)) the `search` input has been refactored to integrate a flat `value` object:

```gql
input search<Model>Input {
  field: AccessionField
  value: String
  valueType: InputType
  operator: Operator
  search: [search<Model>Input]
}
```

This issue proposes a parallel refactor in the `single-page-app-codegen`, to keep `spa` and `graphql-server` changes in sync with each other.

### PR DESCRIPTION
**Pull request by davelsan**
## Issue

Closes #60.
Affects `graphql-server-model-codegen` [issue#140](https://github.com/Zendro-dev/graphql-server-model-codegen/issues/140).

## Description

This PR implements the `single-page-app-codegen` changes necessary to address `graphql-server-model-codegen` [issue#140](https://github.com/Zendro-dev/graphql-server-model-codegen/issues/140).

It also fixes `single-page-app-codegen` integration tests to work with the latest configuration changes introduced in release [0.3.0](https://github.com/Zendro-dev/single-page-app-codegen/releases/tag/0.3.0).

## Changes

- Refactored [model.ejs](https://github.com/Zendro-dev/single-page-app-codegen/commit/7c6ab7e3a45885cfe359dc4764e3f7933499634a#diff-0fb86b1342abeca605f0979941780580) to use the new search value spec.
- Adapted integration tests to use the new `graphql-server` configuration.

## Blocks

None. This PR should be merged simultaneously with:

- `graphql-server` PR [#44](https://github.com/Zendro-dev/graphql-server/pull/44).
- `graphql-server-model-codegen` PR [#144](https://github.com/Zendro-dev/graphql-server-model-codegen/pull/144)

### CODE REVIEW

**Comment by davelsan** [comment]
Testing branches are set to the related `graphql-server` and `graphql-server-model-codegen` feature branches. We need to revert these back to `master` before merging.

File: docker/Dockerfile.graphql_server [hunk 1]
```diff
@@ -5,7 +5,7 @@ WORKDIR /usr/src/app
 
 # Clone the skeleton project and install dependencies
 RUN apt-get update && apt-get install -y git procps autoconf libtool make &&\
-  git clone --branch master https://github.com/Zendro-dev/graphql-server.git . && \
+  git clone --branch codegen#140-refactor-search-input https://github.com/Zendro-dev/graphql-server.git . && \
```
Comment (framirez07) [inline docker/Dockerfile.graphql_server:8]: When doing the merge to master branch we will need to change this to 'master

File: test/integration-test.sh [hunk 2]
```diff
@@ -140,7 +140,7 @@ GQL_TEST_MODELS_INSTANCE2="./test/integration_test_models_instance2"
 TARGET_DIR="./docker/integration_test_run"
 GQL_CODEGEN_DIR="./docker/graphql-server-model-codegen"
 GQL_CODEGEN_URL="https://github.com/Zendro-dev/graphql-server-model-codegen.git"
-GQL_CODEGEN_BRANCH_TAG="master"
+GQL_CODEGEN_BRANCH_TAG="issue#140-refactor-search-input"
```
Comment (framirez07) [inline test/integration-test.sh:143]: When merge to master this will need to change to 'master' branch.

**Review by framirez07 — APPROVED** [review]
Ok

[comment](https://github.com/Zendro-dev/graphql-server/pull/44#issuecomment-690871782)
[comment](https://github.com/Zendro-dev/graphql-server/pull/44#pullrequestreview-486473596)

**Pull request merged**